File: Codewars/4kyu/roman-numerals-encoder/C++/solutions2.cpp
string solution(int n) {
    // 列出所有可能的進位值
    map<int, string> symbols = {
        {1000, "M"}, 
        {900, "CM"}, {500, "D"}, {400, "CD"}, {100, "C"}, 
        {90, "XC"}, {50, "L"}, {40, "XL"}, {10, "X"}, 
        {9, "IX"}, {5, "V"}, {4, "IV"}, {1, "I"}
    };
    
    string result = "";
    map<int, string>::reverse_iterator rit = symbols.rbegin();
    for (; (rit != symbols.rend()) && (n > 0); ++rit) {
        // 符號出現次數
        for (int i = 0; i < (n / rit->first); ++i) {
            result += rit->second;
        }
        // 剩餘位數
        n %= rit->first;
    }
        
    return result;
}

File: Codewars/4kyu/roman-numerals-encoder/JavaScript/solutions2.js
// JavaScript - Node v6.11.0

function solution(n) {
    // 列出所有可能的進位值
    var symbols = {
        M: 1000, 
        CM: 900, D: 500, CD: 400, C: 100, 
        XC: 90, L: 50, XL: 40, X: 10, 
        IX: 9, V: 5, IV: 4, I: 1
    }
    
    var result = "";
    for (var r in symbols) {
        if (n <= 0) {
            break;
        }
        // 符號出現次數
        for (var i = 0; i < parseInt(n / symbols[r]); ++i) {
            result += r;
        }
        // 剩餘位數
        n %= symbols[r];
    }
        
    return result;
}